src/database-layer/db.ts

```diff
@@ -710,6 +717,27 @@ if (maybePg != null) {
 		return {
 			engine: Engines.postgres,
 			executeSql: atomicExecuteSql,
+			on: async (name, fn, options) => {
+				if (name !== 'notification') {
+					throw new Error(`Unsupported listener type: ${name}`);
+				} else if (options?.channel == null) {
+					throw new Error('Missing channel option for notification listener');
+				} else if (options.channel.includes('"')) {
+					throw new Error(
+						`Invalid channel name for task LISTEN: ${options.channel}`,
+					);
+				}
+
+				const client = await pool.connect();
```

Comment:
This will take a client from the pool for every notification listener that is added which could potentially very quickly exhaust the pool, I think a better approach would be to use
```suggestion
				listenerClient ??= await pool.connect();
```
with a `let listenerClient;` in the closure scope above, so that it's possible to reuse a single listener client. Along with adding an `if (msg.channel !== options.channel) return;` to the callback